Build failure: clisp

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Standard build from system packages.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/302524446

### Relevant log output

```console
Note that there are actually two different build failures. 

The one I had is probably "just" a missing dependency:

error: builder for '/nix/store/1a2q3xyf68klfa4c0vqhhrp2igmacp7p-clisp-2.49.95-unstable-2024-12-28.drv' failed with exit code 2;
       last 25 log lines:
       >     make check
       > # Test non-portable features and modules
       >     make extracheck mod-check
       > configurePhase completed in 35 seconds
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash
       > ln -s ../src/.gdbinit .gdbinit
       > rm -f build-aux
       > ln -s ../src/build-aux build-aux
       > mkdir -p gllib
       > test -f gllib/Makefile || sh config.status gllib/Makefile depfiles
       > cd gllib && make CFLAGS=" -g -O2 -no-integrated-cpp -W -Wswitch -Wcomment -Wpointer-arith -Wreturn-type -Wmissing-declarations -Wimplicit -Wno-sign-compare -Wno-format-nonliteral -Wno-shift-negative-value -O -fwrapv -pthread -fno-strict-aliasing -DNO_ASM -DENABLE_UNICODE -DMULTITHREAD -DPOSIX_THREADS -DDYNAMIC_FFI -DDYNAMIC_MODULES  -fPIC -DPIC " top_srcdir="/tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/src"
       > make[1]: Entering directory '/tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/builddir/gllib'
       > cd .. && make  am--refresh
       > make[2]: Entering directory '/tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/builddir'
       > make[2]: Nothing to be done for 'am--refresh'.
       > make[2]: Leaving directory '/tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/builddir'
       >  cd /tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/src && /nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash /tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/src/build-aux/missing automake-1.17 --gnits gllib/Makefile
       > /tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/src/build-aux/missing: line 85: automake-1.17: command not found
       > WARNING: 'automake-1.17' is missing on your system.
       >          You should only need it if you modified 'Makefile.am' or
       >          'configure.ac' or m4 files included by 'configure.ac'.
       > make[1]: *** [Makefile:3188: ../../src/gllib/Makefile.in] Error 127
       > make[1]: Leaving directory '/tmp/nix-build-clisp-2.49.95-unstable-2024-12-28.drv-0/source/builddir/gllib'
       > make: *** [Makefile:654: gllib/libgnu.a] Error 2
       For full logs, run:
         nix log /nix/store/1a2q3xyf68klfa4c0vqhhrp2igmacp7p-clisp-2.49.95-unstable-2024-12-28.drv


The Hydra one is a PathELF failure:


Running phase: fixupPhase
shrinking RPATHs of ELF executables and libraries in /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/bin/clisp
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/full/lisp.run
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/full/modules.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/base/readline.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/base/modules.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/base/regexi.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/base/gettext.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/base/lisp.run
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/base/calls.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/dynmod/lib-rawsock.so
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/dynmod/lib-clx.so
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/dynmod/lib-linux.so
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/dynmod/lib-pcre.so
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/dynmod/lib-zlib.so
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/pcre/cpcre.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/bindings/glibc/linux.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/zlib/zlib.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/rawsock/rawsock.o
patchelf: wrong ELF type
shrinking /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/clx/new-clx/clx.o
patchelf: wrong ELF type
checking for references to /build/ in /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28...
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
patchelf: wrong ELF type
gzipping man pages under /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/share/man/
patching script interpreter paths in /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28
/nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/bin/clisp-link: interpreter directive changed from "#!/bin/sh" to "/nix/store/345523198bcsdzay55pfiimkiajq6lq8-bash-interactive-5.2p37/bin/sh"
stripping (with command strip and flags -S -p) in  /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/bin
ERROR: noBrokenSymlinks: the symlink /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/full/cpcre.o points to /build directory: /build/source/builddir/pcre/cpcre.o
ERROR: noBrokenSymlinks: the symlink /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/full/linux.o points to /build directory: /build/source/builddir/bindings/glibc/linux.o
ERROR: noBrokenSymlinks: the symlink /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/full/clx.o points to /build directory: /build/source/builddir/clx/new-clx/clx.o
ERROR: noBrokenSymlinks: the symlink /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/full/zlib.o points to /build directory: /build/source/builddir/zlib/zlib.o
ERROR: noBrokenSymlinks: the symlink /nix/store/0pjm7l4w8jnczdvc71xk2npqls4v3xzg-clisp-2.49.95-unstable-2024-12-28/lib/clisp-2.49.95+/full/rawsock.o points to /build directory: /build/source/builddir/rawsock/rawsock.o
ERROR: noBrokenSymlinks: found 5 dangling symlinks, 0 reflexive symlinks and 0 unreadable symlinks
```

### Additional context

_No response_

### System metadata

󰳠 nix-shell -p nix-info --run "nix-info -m"                                                                                                                                                                                                                                                                     at 09:04:11
 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.5, NixOS, 25.11 (Xantusia), 25.11.20250708.9807714`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `""`
 - nixpkgs: `/nix/store/bgl6ldj5ihbwcq8p42z3a0qzgqafgk2b-source`

### Notify maintainers

@7c6f434c @lukego @nagy @Uthar @hraban

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/interpreters/clisp/default.nix
+++ b/pkgs/development/interpreters/clisp/default.nix
@@ -130,10 +130,16 @@ stdenv.mkDerivation {
 
   doCheck = true;
 
-  postInstall = lib.optionalString (withModules != [ ]) (
-    ''bash ./clisp-link add "$out"/lib/clisp*/base "$(dirname "$out"/lib/clisp*/base)"/full''
-    + lib.concatMapStrings (x: " " + x) withModules
-  );
+  postInstall = lib.optionalString (withModules != [ ]) ''
+    bash ./clisp-link add "$out"/lib/clisp*/base "$(dirname "$out"/lib/clisp*/base)"/full \
+      ${lib.concatMapStrings (x: " " + x) withModules}
+
+    find "$out"/lib/clisp*/full -type l -name "*.o" | while read -r symlink; do
+      if [[ "$(readlink "$symlink")" =~ (.*\/builddir\/)(.*) ]]; then
+        ln -sf "../''${BASH_REMATCH[2]}" "$symlink"
+      fi
+    done
+  '';
 
   env.NIX_CFLAGS_COMPILE = "-O0 -falign-functions=${
     if stdenv.hostPlatform.is64bit then "8" else "4"

